lib/utils.py

```diff
@@ -108,6 +107,26 @@ def set_extras_n_fix_units(
 
     args["resource_provides_quoted"] = [fixquote(x) for x in args["resource_provides"]]
 
+    # if the user defined the usage_model on the command line,
+    # we need to use their definition of usage_model, not ours
+    # also, we define the site as 'Fermigrid' if they have not requested OFFSITE
+    add_site = ""
+    for r in args["resource_provides_quoted"]:
+        if "usage_model" in r:
+            args["usage_model"] = ""
+            if "OFFSITE" not in r:
+                add_site = "Fermigrid"
+
+    # if the user chooses 'onsite' from the runtime params
+    # we need to define the sites as 'Fermigrid'
+    if args["usage_model"] != "" and "OFFSITE" not in args["usage_model"]:
+        add_site = "Fermigrid"
+
+    if args["site"]:
```

Comment:
So I missed this in my last review, but thankfully our unit tests caught this!  This line will fail with a `KeyError` if `--site` is not set by the user.  What this line needs to be is something that assumes that `args["site"]` may not be set at all.  So either 
```python
if args.get("site", None):
``` 
or
```python
try:
    if args["site"]:
    args["site"] += ", " + add_site
except KeyError:
    args["site"] = add_site
```
or something like that.

Reply:
fixed